source/Irrlicht/CImageLoaderTGA.cpp

```diff
@@ -138,7 +139,7 @@ IImage* CImageLoaderTGA::loadImage(io::IReadFile* file) const
 	if (header.ColorMapType)
 	{
 		// Create 32 bit palette
-		const irr::u16 paletteSize = core::max_((u16)256, header.ColorMapLength);	// ColorMapLength can lie, but so far we only use palette for 8-bit, so ensure it has 256 entries
+		const irr::u16 paletteSize = MAX((u16)256u, (u16)(header.ColorMapType)); // ColorMapLength can lie, but so far we only use palette for 8-bit, so ensure it has 256 entries
```

Comment:
I'd simply use `std::max` (`#include <algorithm>`) for this. You should be able to write `std::max<u16>(256, header.ColorMapType)` and have it work.

Reply:
According to here https://en.cppreference.com/w/cpp/algorithm/max, `std::max` is defined as:

```
const T& max( const T& a, const T& b );
```

This means that we would have the same problem if we use `std::max`. Or, am I missing something here?

Reply:
`ColorMapLength` is already `u16`. Why do you want to cast that to `u16`?

`256` is in principle an `int` literal. But when we pass it to a function which expects `u16`, it will be converted; since it fits both in an `int` (without the sign bit being set) and in `u16`, all is fine. I don't see the issue here. Though feel free to keep the `u` suffix if you prefer that.

(Also, I just see that `ColorMapLength` mutated to `ColorMapType`. Please fix that.)

Reply:
> (Also, I just see that ColorMapLength mutated to ColorMapType. Please fix that.)

Thanks for commenting this! I missed that part, will fix it now.

> ColorMapLength is already u16. Why do you want to cast that to u16?

Yeah it is, no casting is needed, I mistakenly referred to `ColorMapType` there instead, which happened during synchronizing my development branch and the testing branch. 

>  I don't see the issue here. Though feel free to keep the u suffix if you prefer that.

The problem seems that since references are implemented using pointers, we can not pass a member of a packed struct because the address may not be aligned.

Reply:
Ah I see, I wasn't aware that the struct was packed (frankly, I would say this is hacky and error-prone and should be refactored eventually, but not in this PR). I would recommend explaining this in a comment.

Reply:
I added a comment, thanks for reviewing!

Reply:
Thank you for the testing & PR!